thread/common/omrthread.c

```diff
@@ -1573,11 +1573,16 @@ thread_wrapper(WRAPPER_ARG arg)
 				affinityCount += 1;
 			}
 		}
-		/* If the affinityCount is zero, this call will have the effect of clearing any affinity that the thread may have inherited
-		 * and it assuming the initial affinity that the process had (if reverting to that affinity isn't possible on the platform,
-		 * the call is supposed to do nothing).
+		/* if J9THREAD_LIB_FLAG_DISABLE_DEFAULT_AFFINITY is set we only want to modify the affinity if
+		 * the caller provided us with a non-zero affinity
 		 */
-		omrthread_numa_set_node_affinity_nolock(thread, affinity, affinityCount, 0);
+		if (affinityCount != 0 || OMR_ARE_NO_BITS_SET(libs->flags, J9THREAD_LIB_FLAG_DISABLE_DEFAULT_AFFINITY)) {
```

Comment:
Shouldn't `J9THREAD_LIB_FLAG_DISABLE_DEFAULT_AFFINITY ` be `J9THREAD_LIB_FLAG_NO_DEFAULT_AFFINITY`?